src/main/java/org/example/bookweb/service/order/OrderServiceImpl.java

```diff
@@ -0,0 +1,121 @@
+        newOrder.setShippingAddress(orderRequestDto.getShippingAddress());
+        newOrder.setStatus(Order.Status.NEW);
+        newOrder.setOrderDate(LocalDateTime.now());
+        newOrder.setTotal(BigDecimal.ZERO);
+        return orderRepository.save(newOrder);
+    }
+
+    @Transactional
+    @Override
+    public OrderResponseDto placeOrder(User user, OrderRequestDto orderRequestDto) {
+        ShoppingCart userCart = shoppingCartRepository.findShoppingCartByUserId(user.getId());
+        User userDb = userRepository.findById(user.getId()).get();
+        Order newSavedOrder = createNewOrder(userDb, orderRequestDto);;
+
+        Set<OrderItem> orderItems = userCart.getCartItems().stream()
+                .map(cartItem -> {
+                    OrderItem orderItem = orderItemMapper.toOrderItem(cartItem);
+                    orderItem.setPrice(cartItem.getBook().getPrice()
+                            .multiply(BigDecimal.valueOf(cartItem.getQuantity())));
+                    orderItem.setOrder(newSavedOrder);
+                    return orderItem;
+                }).collect(Collectors.toSet());
+        orderItemRepository.saveAll(orderItems);
+
+        newSavedOrder.setTotal(
+                orderItems.stream()
+                .map(OrderItem::getPrice)
+                .reduce(BigDecimal.ZERO, BigDecimal::add)
+        );
+        newSavedOrder.setOrderItems(orderItems);
+
+        shoppingCartRepository.deleteById(userCart.getId());
+        shoppingCartService.createShoppingCart(userDb);
+        return orderMapper.toDto(newSavedOrder);
+    }
+
+    @Override
+    public List<OrderResponseDto> getOrdersByUser(User user) {
+        return orderMapper.toDto(orderRepository.findByUserId(user.getId()));
+    }
+
+    @Override
+    public OrderResponseDto updateStatusByOrderId(Long orderId,
+                                                  OrderStatusUpdateRequest statusUpdateRequest) {
+        Order.Status statusEnum = Order.Status.fromString(statusUpdateRequest.getStatus());
+
+        Order order = orderRepository.findById(orderId).orElseThrow(
+                () -> new EntityNotFoundException("Can't find order by id: " + orderId)
+        );
+
+        order.setStatus(statusEnum);
+        Order savedOrder = orderRepository.save(order);
+        return orderMapper.toDto(orderRepository.getOrderById(savedOrder.getId()));
+    }
+
+    @Override
+    public Set<OrderItemResponseDto> getOrderItemsByOrderId(Long orderId, User user) {
+        return orderItemMapper.toDto(
+                orderItemRepository.findOrderItemsByOrderIdAndUserId(orderId, user.getId())
+        );
+    }
+
+    @Override
+    public OrderItemResponseDto getOrderItemByIdAndOrderId(User user,
+                                                           Long orderId,
+                                                           Long orderItemId) {
+        return orderItemMapper.toDto(orderItemRepository.findOrderItemByIdAndOrderIdAndUserId(
```

Comment:
To increase readability let's use
```suggestion
        OrderItem item = orderItemRepository.findByIdAndOrderIdAndUserId(orderItemId, orderId, userId)
                .orElseThrow(() -> new EntityNotFoundException(
                        String.format("Order item: %d not found in order: %d for user: %d",
                                orderItemId, orderId, userId)
                ));
        return orderItemMapper.toOrderItemDto(item);
```